src/Entities/Entity.cpp

```diff
@@ -1134,6 +1135,14 @@ void cEntity::HandlePhysics(std::chrono::milliseconds a_Dt, cChunk & a_Chunk)
 				// We hit the ground, adjust the position to the top of the block:
 				m_bOnGround = true;
 				NextPos.y = HitBlockCoords.y + 1;
+
+				// If this is a pickup and it hits the ground then stop it
+				if (GetEntityType() == eEntityType::etPickup)
+				{
+					NextSpeed.x = 0.0;
+					NextSpeed.y = 0.0;
+					NextSpeed.z = 0.0;
+				}
```

Comment:
I really don't like having specific code in the generic function. Wouldn't it be possible to move this into `cPickup`'s override of the `Tick()` method?